common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -883,6 +918,25 @@ def answer_available(self):
 
         return False
 
+    def correctness_available(self):
+        """
+        Is the user allowed to see whether she's answered correctly?
+
+        Limits access to the correct/incorrect flags, messages, and problem score.
+        """
+        if self.show_correctness == SHOW_CORRECTNESS.NEVER:
+            return False
+        elif self.runtime.user_is_staff:
+            # This is after the 'never' check because admins can see correctness
+            # unless the problem explicitly prevents it
+            return True
+        elif self.show_correctness == SHOW_CORRECTNESS.PAST_DUE:
+            return self.is_past_due()
+        elif self.show_correctness == SHOW_CORRECTNESS.ALWAYS:
+            return True
+
+        return True
```

Comment:
I like having a final catch-all return, but agree it's a little redundant.

How about 3f23ea15e6 instead?